src/plugins/score-plugin-packagemanager/PackageManager/View.cpp

```diff
@@ -213,155 +239,224 @@ void PluginSettingsView::on_message(QNetworkReply* rep)
+        addon.file,
+        QDir{installPath}.absolutePath(),
+        [=](const std::vector<QString>& res) {
+    reset_progress();
+    if (res.empty())
+      return;
+    // We want the extracted folder to have the name of the addon
+    {
+      QDir addons_dir{installPath};
+      QFileInfo a_file(res[0]);
+      auto d = a_file.dir();
+      auto old_d = d;
+      while (d.cdUp() && !d.isRoot())
+      {
+        if (d == addons_dir)
         {
-          QDir addons_dir{installPath};
-          QFileInfo a_file(res[0]);
-          auto d = a_file.dir();
-          auto old_d = d;
-          while (d.cdUp() && !d.isRoot())
-          {
-            if (d == addons_dir)
-            {
-              addons_dir.rename(old_d.dirName(), addon.raw_name);
-              break;
-            }
-            old_d = d;
-          }
+          addons_dir.rename(old_d.dirName(), addon.raw_name);
+          break;
         }
+        old_d = d;
+      }
+    }
 
-        QMessageBox::information(
-            m_widget,
-            tr("Addon downloaded"),
-            tr("The addon %1 has been succesfully installed in :\n"
-               "%2\n\n"
-               "It will be built and enabled shortly.\nCheck the message "
-               "console for errors if nothing happens.")
-                .arg(addon.name)
-                .arg(QFileInfo(installPath).absoluteFilePath()));
-      },
-      [=] {
-        m_progress->setHidden(true);
-        QMessageBox::warning(
-            m_widget,
-            tr("Download failed"),
-            tr("The package %1 could not be downloaded.").arg(addon.name));
-      });
+    QMessageBox::information(
+          m_widget,
+          tr("Addon downloaded"),
+          tr("The addon %1 has been succesfully installed in :\n"
+             "%2\n\n"
+             "It will be built and enabled shortly.\nCheck the message "
+             "console for errors if nothing happens.")
+          .arg(addon.name)
+          .arg(QFileInfo(installPath).absoluteFilePath()));
+  },
+  [=](qint64 recieved, qint64 total) { progress_from_bytes(recieved, total); },
+  [=] {
+    reset_progress();
+    QMessageBox::warning(
+          m_widget,
+          tr("Download failed"),
+          tr("The package %1 could not be downloaded.").arg(addon.name));
+  });
 }
 
-void PluginSettingsView::installSDK(const RemotePackage& addon)
+void PluginSettingsView::installSDK(const Package& addon)
 {
-  QString version = SCORE_TAG_NO_V;
   constexpr const char* platform =
-#if defined(_WIN32)
+    #if defined(_WIN32)
       "windows"
-#elif defined(__APPLE__)
+    #elif defined(__APPLE__)
       "mac"
-#else
+    #else
       "linux"
-#endif
+    #endif
       ;
 
   const QString sdk_path{
-      score::AppContext().settings<Library::Settings::Model>().getSDKPath() + "/" + version};
+    score::AppContext().settings<Library::Settings::Model>().getSDKPath() + '/' + SCORE_TAG_NO_V};
```

Comment:
better to keep using version here, it saves a string -> qstring conversion

Reply:
Strange, I cannot find this in my code
https://github.com/thibaudk/score/blob/wip/pkgMng/src/plugins/score-plugin-packagemanager/PackageManager/View.cpp#L406